### PR TITLE
Enabled object detection with YOLO model

### PR DESCRIPTION
**Pull request by aozo**
### Summary of Changes

* Modified e2e.conf file to load a yolo3_mobilenet_lite model trained on the CARLA data set with only traffic signs.
* Copied YOLO preprocessing and postprocessing files from the https://github.com/abrahamchan/keras-YOLOv3-model-set repository to output the object detection outputs in the format that Pylot expects.
* Modified the `split_dataset.py` script to keep class ID 0 reserved so that the split dataset adheres to the COCO dataset format.

### Validation
* Confirmed that object detection is working E2E with the YOLO model and the speed is adjusted accordingly.

### CODE REVIEW

File: configs/e2e.conf [hunk 1]
```diff
@@ -40,16 +45,18 @@
 --camera_image_width=1280
 --camera_image_height=720
 --simulator_town=1
+# Spawns near a few speed signs
+--simulator_spawn_point_index=108
```
Comment (aozo) [inline configs/e2e.conf]: Found that this flag allows us to hard-code spawn points in a map. This one spawns near a few traffic signs so it was helpful for testing.

File: dependencies/models/yolo3_anchors.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
```
Comment (aozo) [inline dependencies/models/yolo3_anchors.txt:1]: Copied from keras-YOLOv3-model-set because it's needed for post-processing the model output.

File: pylot/perception/detection/detection_operator.py [hunk 3]
```diff
@@ -73,14 +77,21 @@ def __init__(self, camera_stream: erdos.ReadStream,
 
         logger.debug('\n\n*********Path of the model using now: %s', model_path)
 
-        self._coco_labels = load_coco_labels(self._flags.path_coco_labels)
+        self._model_names = self._flags.obstacle_detection_model_names
+        if 'yolo' in self._model_names:
+            self._coco_labels = load_coco_labels(self._flags.path_coco_labels, is_zero_index=True)
+        else:
+            self._coco_labels = load_coco_labels(self._flags.path_coco_labels)
```
Comment (aozo) [inline pylot/perception/detection/detection_operator.py:84]: Pylot assumes all object detectors will output class IDs that are 1-indexed. However, keras-YOLOv3-model-set trains the model with classes being 0-indexed.

File: pylot/perception/detection/detection_operator.py [hunk 4]
```diff
@@ -217,3 +232,38 @@ def __run_model(self, image_np):
         res_boxes = boxes[0][:num_detections]
         res_scores = scores[0][:num_detections]
         return num_detections, res_boxes, res_scores, res_classes
+
+    def __run_yolo_model(self, image_np):
```
Comment (aozo) [inline pylot/perception/detection/detection_operator.py:236]: New inference function specific to YOLO. Some notable differences include:
* YOLO requires inputs to be RGB format, but CARLA outputs frames in BGR format.
* YOLO input tensor shape differs from the models in the Tensorflow Object Detection zoo and the Pylot default object detection models.
* YOLO output bounding boxes are not normalized to the range [0, 1] which Pylot expects.

File: pylot/perception/detection/yolo/common/data_utils.py [hunk 5]
```diff
@@ -0,0 +1,1141 @@
+#!/usr/bin/python3
```
Comment (aozo) [inline pylot/perception/detection/yolo/common/data_utils.py:1]: Copied from keras-YOLOv3-model-set. No changes made.

File: pylot/perception/detection/yolo/common/wbf_postprocess.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/python3
```
Comment (aozo) [inline pylot/perception/detection/yolo/common/wbf_postprocess.py:1]: Copied from keras-YOLOv3-model-set. No changes made.

File: pylot/perception/detection/yolo/common/yolo_postprocess_np.py [hunk 7]
```diff
@@ -0,0 +1,584 @@
+#!/usr/bin/python3
```
Comment (aozo) [inline pylot/perception/detection/yolo/common/yolo_postprocess_np.py:1]: Copied from keras-YOLOv3-model-set. No changes made.

File: pylot/perception/detection/yolo/yolo3/postprocess_np.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
```
Comment (aozo) [inline pylot/perception/detection/yolo/yolo3/postprocess_np.py:1]: Copied from keras-YOLOv3-model-set. No changes made.

File: configs/e2e.conf (hunk 1 above)
Comment (abrahamchan) [reply configs/e2e.conf]: Is there a reference guide for the spawn point index? Is it unique per map?

Comment (aozo) [reply configs/e2e.conf]: The index is unique per map and I'm not aware of any reference guide. This particular index was determined experimentally by printing out the index.

Comment (aozo) [reply configs/e2e.conf]: As discussed, moved this to a separate conf file.

**Comment by abrahamchan** [comment]
LGTM!
